ui/src/message_popup/mobile_prototype/texting_interface.jsx

```diff
@@ -0,0 +1,184 @@
+            return (
+              <VelocityComponent animation='transition.expandIn' runOnMount={true} key={'animation'+ message.key}>
+                <UserMessage text={message.text}  key={message.key} />
+              </VelocityComponent>
+              );
+          }else if(message.type === 'student'){
+            return (
+              <VelocityComponent animation='transition.expandIn' runOnMount={true} key={'animation'+ message.key}>
+                <StudentMessage text={message.text} student={message.student} onOpenStudentDialog={this.props.onOpenStudentDialog(message.student)} key={message.key} />
+              </VelocityComponent>
+              );
+          }else{
+            return (
+              <VelocityComponent animation='transition.expandIn' runOnMount={true} key={'animation'+ message.key}>
+                <InfoMessage text={message.text} onOpenInfoDialog={this.props.onOpenInfoDialog} key={message.key} />
+              </VelocityComponent>
+              );
+          }
+        })}
+      </div>
+    );
+  }
+});
+
+export const TextFooter = React.createClass({
+  displayName: 'TextFooter',
+  
+  propTypes: {
+    onQuestionDone: React.PropTypes.func.isRequired,
+    setInitialResponse: React.PropTypes.func.isRequired,
+    setRevisedResponse: React.PropTypes.func.isRequired,
+    setPassedResponse: React.PropTypes.func.isRequired,
+    log: React.PropTypes.object.isRequired,
+    isReadyToMoveOn: React.PropTypes.func.isRequired,
+    helpType: React.PropTypes.string.isRequired,
+    elapsedMs: React.PropTypes.number.isRequired,
+    nextButtonLabel: React.PropTypes.string.isRequired,
+    nextExample: React.PropTypes.func.isRequired,
+    showExample: React.PropTypes.func.isRequired
+  },
+  
+  getInitialState(){
+    return ({
+      text: undefined,
+      initialMessageSent: false,
+    });
+  },
+  
+  onTextChange(event, value){
+    this.setState({text: value});
+  },
+  
+  onSendMessage(){
+    this.props.setInitialResponse(this.state.text);
+    this.setState({initialMessageSent: true});
+    this.props.log.logResponse();
+  },
+  
+  onRevisionSendMessage(){
+    this.props.setRevisedResponse(this.state.text);
+    this.props.log.logRevision();
+  },
+  
+  onPassSendMessage(){
+    this.props.setPassedResponse('');
+    this.props.log.logRevisionDeclined();
+  },
+  
+  onNextQuestion(){
+    this.props.onQuestionDone();
+  },
+  
+  render(){
+    const mainButtonLabel = this.props.isReadyToMoveOn() ? this.props.nextButtonLabel : this.props.helpType !== 'feedback' ? 'Send' : this.state.initialMessageSent ? 'Revise Response' : 'Save Response';
+    return (
+      <div>
+        <Divider />
+        <div style={styles.responseTextBox}>
+          <TextField 
+            id="Response"
+            multiLine={true}
+            underlineShow={false}
+            rows={2}
+            rowsMax={2}
+            onChange={this.onTextChange}
+            disabled={this.props.isReadyToMoveOn()}
```

Comment:
Instead of passing down the function `isReadyToMoveOn`, pass the value directly.